In [1]:
import pandas as pd

df = pd.read_csv('../data/items_titles_test.csv')

import sys
sys.path.append('..')
from scripts.similarity import TextSimilarityPipeline

# Crear instancia y ejecutar
similitud = TextSimilarityPipeline(df)
resultado = similitud.run_pipeline()

# Mostrar resultado
resultado

Cálculo paralelo completado en 1265.21 segundos
Pipeline ejecutado en 1301.72 segundos


Término 1  \
43505751            Sapato Usthemp Legend Temático - Pug 3   
6212817        Tênis Usthemp Long Temático - Tingo Almeida   
15618237                                     Tênis Slip On    
39891505     Tênis Usthemp Folks Vegano Casual Estampa Olly   
11315110  Tenis Infantil Tênis Feminino Promoção Brinde ...   
...                                                     ...   
12657526                               Tenis Try Way - Novo   
31557414                   16 Centros De Mesa Gatinha Marie   
10501311                                    Btt De 2 Partes   
48076696                Tenis Comfortflex 21-62305 Feminina   
37436991  Sandália Anabela Azaleia  Maxi Fivela Feminina...   

                                                  Término 2  Similitud  
43505751            Sapato Usthemp Legend Temático - Pug 4   1.000000  
6212817       Tênis Usthemp Long Temático - Jardim Secreto   1.000000  
15618237                                      Tênis Slip On   1.000000  
39891505  Tênis Usthemp Folks Vegano Casual Estampa Pequ...   1.000000  
11315110  Tenis Infantil Tênis Feminino Promoção Brinde ...   1.000000  
...                                                     ...        ...  
12657526                                               Ês    -0.483827  
31557414                                               Piso  -0.488728  
10501311  Bike 26 Elleven 21v Freios A Disco Câmbios Shi...  -0.508922  
48076696                                            Bicleta  -0.518127  
37436991                                            Bicleta  -0.533344  

[49995000 rows x 3 columns]

In [2]:
df.shape

(10000, 1)

## Análisis de Resultados de Similitud de Texto

### Detalles del Dataset
- **Cantidad de textos analizados:** 10,000

### Tiempos de Ejecución
1. **Cálculo paralelo de similitudes:** 1265.21 segundos (~21 minutos)
2. **Tiempo total del pipeline:** 1301.72 segundos (~22 minutos)

### Observaciones
- El uso de **Truncated SVD** para reducir la dimensionalidad, combinado con paralelización, permitió calcular similitudes entre todos los pares de textos en un tiempo razonable para el tamaño del dataset.
- Este enfoque es escalable y puede manejar datasets de gran tamaño con optimizaciones adicionales.